lib/connection/connection_config.js

```diff
@@ -260,17 +221,29 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
         protocol: proxyProtocol,
         noProxy: noProxy
       };
-    } else {
-      proxy =
-      {
-        host: proxyHost,
-        port: proxyPort,
-        protocol: proxyProtocol,
-        noProxy: noProxy
-      };
+    } 
+  
+    if (!proxy && GlobalConfig.isEnvProxyActive()) {
```

Comment:
what happens if the user deliberately configures `proxyHost` + `proxyPort` (optonally user + password) on the connection config, and their environment also has HTTP_PROXY or HTTPS_PROXY ?

Reply:
Then, the driver will only use the proxy info in the connection option,